integration/openshift/e2e/gen_test_cases.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+#
+# Copyright (c) 2020 Red Hat, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# This script should be used to generate the list of test cases from
+# the OpenShift e2e conformance suite.
+#
+# Edit the filter_patterns, exclude_patterns, and skip_patterns variables
+# and re-run this script in order to change the tests selection.
+#
+set -e
+
+GOPATH=${GOPATH:-$(go env GOPATH)}
+script_dir="$(realpath $(dirname $0))"
+source "$script_dir/../../../lib/common.bash"
+tests_dir="$script_dir/../../../"
+source "$script_dir/../lib.sh"
+
+# Tests that should be included. Leave it empty to pick them all.
+filter_patterns=(
```

Comment:
Hi @c3d , hey, thanks for reviewing too!

> I would not call them patterns.

How would you call them? I'm really expecting grep-like (extended) patterns in quotes as you mentioned.

> 
> The problem is that there are two expansions at play here, the first one being the shell, the second one from your `grep` below. So if you have `*.znort` here, this will really be expanded by the shell first.
> 

I didn't understand the second expansion you mentioned. Because It writes the patterns to a file, then use the `grep -f` to read them from the same file.
 
> Maybe having a commended-out example of the kind of pattern you expect here? I suspect it would be grep-like patterns in quotes, i.e. `'foo*'` would not match `fooa` (not glob) but would match `foooo` (because `grep`).

Yeah, it deserves a better text explaining that.

Reply:
> Hi @c3d , hey, thanks for reviewing too!
> 
> > I would not call them patterns.
> 
> How would you call them? I'm really expecting grep-like (extended) patterns in quotes as you mentioned.

Sorry, should have been clear. I would call them "regexps", that way you know it's not glob patterns.

> 
> > The problem is that there are two expansions at play here, the first one being the shell, the second one from your `grep` below. So if you have `*.znort` here, this will really be expanded by the shell first.
> 
> I didn't understand the second expansion you mentioned. Because It writes the patterns to a file, then use the `grep -f` to read them from the same file.

If in your script someone writes 

```
filter_patterns=(
    *.txt
)
```

then the shell first interprets that as a glob pattern, so you end up with an array containing maybe something like `a.txt`, `b.txt`, and so on.


> 
> > Maybe having a commended-out example of the kind of pattern you expect here? I suspect it would be grep-like patterns in quotes, i.e. `'foo*'` would not match `fooa` (not glob) but would match `foooo` (because `grep`).
> 
> Yeah, it deserves a better text explaining that.